components/camel-zeebe/src/main/java/org/apache/camel/component/zeebe/processor/DeploymentProcessor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.zeebe.processor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.camel.CamelException;
+import org.apache.camel.Exchange;
+import org.apache.camel.component.zeebe.ZeebeConstants;
+import org.apache.camel.component.zeebe.ZeebeEndpoint;
+import org.apache.camel.component.zeebe.internal.ZeebeService;
+import org.apache.camel.component.zeebe.model.DeploymentRequest;
+import org.apache.camel.component.zeebe.model.DeploymentResponse;
+import org.apache.camel.component.zeebe.model.ProcessDeploymentResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DeploymentProcessor extends AbstractBaseProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(DeploymentProcessor.class);
+
+    public DeploymentProcessor(ZeebeEndpoint endpoint) {
+        super(endpoint);
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        DeploymentRequest message = null;
+
+        Object body = exchange.getMessage().getBody();
+        String headerResourceName = exchange.getMessage().getHeader(ZeebeConstants.RESOURCE_NAME, String.class);
+        if (headerResourceName != null && (body instanceof String || body instanceof byte[])) {
+            message = new DeploymentRequest();
+            message.setName(headerResourceName);
+            if (body instanceof String) {
+                message.setContent(((String) body).getBytes());
+            } else {
+                message.setContent((byte[]) body);
+            }
+        } else if (body instanceof DeploymentRequest) {
+            message = (DeploymentRequest) body;
+        } else if (body instanceof String) {
+            try {
+                message = objectMapper.readValue((String) body, DeploymentRequest.class);
+            } catch (JsonProcessingException jsonProcessingException) {
+                throw new IllegalArgumentException("Cannot convert body to DeploymentRequestMessage", jsonProcessingException);
+            }
+        } else {
```

Comment:
The body can be inputstream and still possible to read into a String so without using camel type converter then this processor is very limited